nulib/dc-nextjs
# 4155 embedded viewer query params

## What does this do?
Supports user control of displaying certain Clover `Viewer` elements for embedded players.   The user can supply query params at the end of the embedded URL, which will get picked up by the embedded player and communicated to Clover `Viewer`.

Here are the current Clover Viewer config options:

```ts
export type ViewerConfigOptions = {
	background?: string;
	canvasBackgroundColor?: string;
	canvasHeight?: string;
	ignoreCaptionLabels?: string[];
	informationPanel?: {
		open?: boolean;
		renderAbout?: boolean;
		renderSupplementing?: boolean;
		renderToggle?: boolean;
	};
	openSeadragon?: OpenSeadragonOptions;
	requestHeaders?: IncomingHttpHeaders;
	showIIIFBadge?: boolean;
	showTitle?: boolean;
	withCredentials?: boolean;
};
```

This issue supports `showTitle`, `informationPanel.open` and `informationPanel.renderToggle`.

One caveat is that the query params must be `Uri Encoded`.  The way the embedding display is wired-up, we can't just let the user append something like `url....?showTitle=false&informationPanelOpen=true` Unless we trust a user to correctly manually `encodeUri` the params, we need some type of helper feature from which they can copy/paste.  So I built some select drop downs which will update the iFrame `src`, and drafted up a quick visual:

<img width="943" alt="image" src="https://github.com/nulib/dc-nextjs/assets/3020266/41e6495c-b0fa-40b4-9a0f-373282f6e1b0">

<img width="941" alt="image" src="https://github.com/nulib/dc-nextjs/assets/3020266/166818ec-3447-45ee-bc76-7ea6b6383d1c">


Also as part of this issue I broke up the `DownloadAndShare` component into more manageable sub-components.

## How to test?
1. Go to a Work page and click "Download and Share"
2. In the Embed Viewer helper, select different options
3. Copy and past the `<iframe src=[COPY ME] />` url and drop that into a fresh browser tab.  The route should respect and render your selected options.
